packages/firestore/src/model/values.ts

```diff
@@ -185,6 +189,11 @@ function objectEquals(left: api.Value, right: api.Value): boolean {
   return true;
 }
 
+/** Returns true if the Value list contains the specified element. */
+export function contains(haystack: api.ArrayValue, needle: api.Value): boolean {
+  return (haystack.values || []).find(v => equals(v, needle)) !== undefined;
+}
+
 export function compare(left: api.Value, right: api.Value): number {
```

Comment:
Reading through calling code I fear that names like `compare` and `equals` are too generic for free functions that don't operate on general arguments.

What do you think of naming these functions with a type-operation naming scheme? This would be `valueCompare`, etc.

WDYT?

Reply:
Yeah, we can do that. I looked into removing the namespace import for our logging already, and `logWarn` was what I picked there as a temporary solution.